front/src/modules/ui/table/components/EntityTableHeaderOptions.tsx

```diff
@@ -0,0 +1,96 @@
+import { useTheme } from '@emotion/react';
+import styled from '@emotion/styled';
+
+import { DropdownButton } from '@/ui/dropdown/components/DropdownButton';
+import { StyledDropdownMenu } from '@/ui/dropdown/components/StyledDropdownMenu';
+import { StyledDropdownMenuItemsContainer } from '@/ui/dropdown/components/StyledDropdownMenuItemsContainer';
+import { useDropdownButton } from '@/ui/dropdown/hooks/useDropdownButton';
+import { ViewFieldMetadata } from '@/ui/editable-field/types/ViewField';
+import {
+  IconArrowNarrowLeft,
+  IconArrowNarrowRight,
+  IconEyeOff,
+} from '@/ui/icon';
+import { MenuItem } from '@/ui/menu-item/components/MenuItem';
+
+import { ColumnHeaderDropdownId } from '../constants/ColumnHeaderDropdownId';
+import { useTableColumns } from '../hooks/useTableColumns';
+import { ColumnDefinition } from '../types/ColumnDefinition';
+
+const StyledDropdownContainer = styled.div`
+  left: 0px;
+  position: absolute;
+  top: 32px;
+  z-index: 1;
+`;
+type OwnProps = {
+  column: ColumnDefinition<ViewFieldMetadata>;
+  isFirstColumn: boolean;
+  isLastColumn: boolean;
+};
+export const EntityTableHeaderOptions = ({
+  column,
+  isFirstColumn,
+  isLastColumn,
+}: OwnProps) => {
+  const theme = useTheme();
+
+  const {
+    handleColumnVisibilityChange,
+    handleColumnLeftMove,
+    handleColumnRightMove,
+  } = useTableColumns();
+
+  const { closeDropdownButton } = useDropdownButton({
+    dropdownId: ColumnHeaderDropdownId,
+  });
+
+  const handleColumnMoveLeft = () => {
+    closeDropdownButton();
+    if (isFirstColumn) return;
+    else handleColumnLeftMove(column);
+  };
+
+  const handleColumnMoveRight = () => {
+    closeDropdownButton();
+    if (isLastColumn) return;
+    else handleColumnRightMove(column);
+  };
+
+  const handleColumnVisibility = () => {
+    handleColumnVisibilityChange(column);
+  };
+
+  return (
+    <DropdownButton
+      dropdownId={ColumnHeaderDropdownId}
+      dropdownComponents={
+        <StyledDropdownContainer>
+          <StyledDropdownMenu>
+            <StyledDropdownMenuItemsContainer>
+              <MenuItem
+                LeftIcon={() => (
+                  <IconArrowNarrowLeft size={theme.icon.size.md} />
+                )}
+                onClick={handleColumnMoveLeft}
+                text="Move left"
+              />
```

Comment:
I am getting a scope error on this, am I missing something to include? 
<img width="1348" alt="Screenshot 2023-09-17 at 7 15 15 PM" src="https://github.com/twentyhq/twenty/assets/13139771/31f707d7-1776-43d0-8c97-8129f0e9b91e">

Reply:
DropdownButton needs a RecoilScope to be present as a parent somewhere in the component hierarchy. It's where it is storing it's state.

So You can add a <RecoilScope CustomRecoilScopeContext={DropdownRecoilScopeContext}> to wrap the DropdownButton. I recommend to put it in the EntityTableHeader component. Then you'll be able to use useDropdownButton hook to open the dropdown on columnHead onClick() for example